File: go-sandbox/tests/math/math.go
// Package math contains tiny numeric helpers exercised by the test suite.
package math

import "errors"

func Add(a, b int) int { return a + b }
func Sub(a, b int) int { return a - b }
func Mul(a, b int) int { return a * b }

// Div returns a/b or an error when b == 0.
func Div(a, b int) (int, error) {
	if b == 0 {
		return 0, errors.New("division by zero")
	}
	return a / b, nil
}

func Factorial(n int) int {
	if n < 0 {
		return 0
	}
	r := 1
	for i := 2; i <= n; i++ {
		r *= i
	}
	return r
}

func Fibonacci(n int) int {
	a, b := 0, 1
	for i := 0; i < n; i++ {
		a, b = b, a+b
	}
	return a
}

func IsPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n%2 == 0 {
		return n == 2
	}
	for i := 3; i*i <= n; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

func GCD(a, b int) int {
	for b != 0 {
		a, b = b, a%b
	}
	return a
}

File: lua-sandbox/test-busted/src/sandbox/math.lua
-- Tiny numeric helpers exercised by the test suite.
local M = {}

function M.add(a, b) return a + b end
function M.sub(a, b) return a - b end
function M.mul(a, b) return a * b end

function M.div(a, b)
  if b == 0 then return nil, "division by zero" end
  return a / b
end

function M.factorial(n)
  if n < 0 then return nil, "negative" end
  local r = 1
  for i = 2, n do r = r * i end
  return r
end

function M.fibonacci(n)
  local a, b = 0, 1
  for _ = 1, n do a, b = b, a + b end
  return a
end

function M.is_prime(n)
  if n < 2 then return false end
  if n % 2 == 0 then return n == 2 end
  local i = 3
  while i * i <= n do
    if n % i == 0 then return false end
    i = i + 2
  end
  return true
end

function M.gcd(a, b)
  while b ~= 0 do a, b = b, a % b end
  return a
end

return M
